pallets/rewards/src/lib.rs

```diff
@@ -0,0 +1,260 @@
+pub enum ItnRewardStatus {
+    Unclaimed(Balance),
+    Claimed,
+}
+
+decl_error! {
+    pub enum Error for Module<T: Config> {
+        /// Address was not found in the list of Itn addresses.
+        UnknownItnAddress,
+        /// Itn reward was already claimed.
+        ItnRewardAlreadyClaimed,
+        /// Provided signature was invalid.
+        InvalidSignature,
+        /// Balance can not be converted to a primitive.
+        UnableToCovertBalance,
+        // Insufficient balance to payout rewards.
+        InsufficientBalance,
+    }
+}
+
+decl_storage! {
+    trait Store for Module<T: Config> as Rewards {
+        /// Map of (Itn Address `AccountId`) -> (Reward `ItnRewardStatus`).
+        pub ItnRewards get(fn itn_rewards): map hasher(blake2_128_concat) T::AccountId
+            => Option<ItnRewardStatus>;
+    }
+    add_extra_genesis {
+        config(itn_rewards): Vec<(T::AccountId, Balance)>;
+        build(|config: &GenesisConfig<T>| {
+            for (account, balance) in &config.itn_rewards {
+                <ItnRewards<T>>::insert(account, ItnRewardStatus::Unclaimed(*balance));
+            }
+        });
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Config> for enum Call where origin: <T as frame_system::Config>::Origin {
+        type Error = Error<T>;
+
+        fn deposit_event() = default;
+
+        /// Claim an ITN reward
+        ///
+        /// ## Arguments
+        /// `itn_address` - Address of the awarded account on ITN.
+        /// `signature` - Signature of the encoded reward data (originKey, ItnKey, amount)
+        ///
+        /// # Errors
+        /// * `InsufficientBalance` - Itn rewards has insufficient funds to issue the reward.
+        /// * `InvalidSignature` - `signature` had an invalid signer or invalid message.
+        /// * `ItnRewardAlreadyClaimed` - Reward issued to the `itn_address` has already been claimed.
+        /// * `UnknownItnAddress` - `itn_address` is not in the rewards table and has no reward to be claimed.
+        #[weight = <T as Config>::WeightInfo::claim_itn_reward()]
+        pub fn claim_itn_reward(origin, reward_address: T::AccountId, itn_address: T::AccountId, signature: T::OffChainSignature) -> DispatchResult {
+            ensure_none(origin)?;
+            <ItnRewards<T>>::try_mutate(&itn_address, |reward| {
+                match reward {
+                    // Unclaimed. Attempt to claim.
+                    Some(ItnRewardStatus::Unclaimed(amount)) => {
+                        let amount = *amount;
+                        // `amount` and `bonded_amount` are equal in value but different types.
+                        // `deposit_amount` is 1 POLY more because we bond `bonded_amount`, we don't want all the users poly bonded.
+                        let (bonded_amount, deposit_amount) = Self::convert_balance(amount)?;
+                        ensure!(
+                            Self::balance() >= deposit_amount,
```

Comment:
Should this not use `T::Currency::ensure_can_withdraw`?

Reply:
The api is a bit odd for `ensure_can_withdraw`, It takes the `amount` and the `new _balance` as args. To get the new balance you have to subtract the amount and so you have to deal with overflows. I think the best way forward is to surface the error from `withdraw` since it calls `ensure_can_withdraw`.